packages/client-engine-runtime/src/interpreter/renderQuery.ts

```diff
@@ -15,16 +27,62 @@ export function renderQuery({ query, params }: QueryPlanDbQuery, scope: ScopeBin
       throw new Error(`Missing value for query variable ${param.prisma__value.name}`)
     }
 
-    return value
+    return value as PrismaValue
   })
+}
+
+function renderTemplateSql(
+  fragments: Fragment[],
+  placeholderFormat: PlaceholderFormat,
+  params: PrismaValue[],
+): SqlQuery {
+  let paramIndex = 0
+  let placeholderNumber = 1
+  const flattenedParams: PrismaValue[] = []
+  const sql = fragments
+    .map((fragment) => {
+      switch (fragment.type) {
+        case 'parameter':
+          flattenedParams.push(params[paramIndex++])
+          return formatPlaceholder(placeholderFormat, placeholderNumber++)
+
+        case 'stringChunk':
+          return fragment.value
+
+        case 'parameterTuple': {
+          const paramValue = params[paramIndex++]
+          const paramArray = Array.isArray(paramValue) ? paramValue : [paramValue]
+          const placeholders =
+            paramArray.length == 0
+              ? 'NULL'
+              : paramArray
+                  .map((value) => {
+                    flattenedParams.push(value)
+                    return formatPlaceholder(placeholderFormat, placeholderNumber++)
+                  })
+                  .join(', ')
+          return `(${placeholders})`
+        }
+
+        default:
+          assertNever(fragment.type, 'Invalid fragment type')
```

Comment:
the function needs to be imported from `../utils.ts`

Reply:
Yeah, I figured it out. Tried to import it from another package, while we had a copy in `utils.ts`. Found it when I wanted to copy it exactly there :)